### PR TITLE
Emma development

### LINKED ISSUE #16: Add Authentication
We want a way for users to be able to save destinations.  In order to do so we will implement the authentication API.

### LINKED ISSUE #20: Verify valid email address
At the moment a user can sign up on our web app using an email and password, however, there is currently no way of telling if the email address is valid.  I would like to add something to verify this?

### LINKED ISSUE #21: Add notifications for invalid requests using authentication
Currently the only way to see if you successfully signed up, logged in or logged out is through looking in the console.  I would like to add a way of notifying the user.

### LINKED ISSUE #31: Hide Firebase API keys
At the moment anyone who views the source code would have access to all the information.  I need to hide it.  I am thinking of adding it to the secrets.json file.

### PR DESCRIPTION
**Pull request by emmapaczkowski**
- I added notifications so that the user is aware of if they have successfully logged in or signed up as well as giving them information as to why it didn't work if it was unsuccessful. Closes #21. Closes #16 
- Verify if the email address input is a valid email format using real expressions.  Closes #20.
- Added the Firebase real time database so that when users first sign up there information is stored in the database. 
- Remove Firebase API and placed it in the secrets.json file. Closes #31

### CODE REVIEW

File: where-to-travel/src/main/webapp/login.js
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/login.js]: loged -> logged

File: where-to-travel/src/main/webapp/signup.js
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js:2]: Prefer to use CONSTANT_CASE: https://google.github.io/styleguide/jsguide.html#naming-constant-names

File: where-to-travel/src/main/webapp/signup.js
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: var -> let

File: where-to-travel/src/main/webapp/signup.js
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: var -> let

File: where-to-travel/src/main/webapp/signup.js
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: var -> let

File: where-to-travel/src/main/webapp/signup.js
Comment (PriyaVarra) [inline where-to-travel/src/main/webapp/signup.js]: add comment explaining what a properly formatted email captured by this regex looks like

**Review by PriyaVarra — COMMENTED** [review]
Just a couple of minor style changes. Otherwise looks good!

File: where-to-travel/src/main/webapp/login.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/login.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js:2]: Should they all be constant case?

Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/signup.js:2]: i believe so

File: where-to-travel/src/main/webapp/login.js
Comment (andy7809) [inline where-to-travel/src/main/webapp/login.js:19]: What is e? Will alerting the user with e help them to complete a successful login?

File: where-to-travel/src/main/webapp/signup.js
Comment (andy7809) [inline where-to-travel/src/main/webapp/signup.js:38]: This line is long, can we split it with a multi line string?

File: where-to-travel/src/main/webapp/signup.js
Comment (wjiamin) [inline where-to-travel/src/main/webapp/signup.js:11]: Should we return an alert/error for this case?

File: where-to-travel/src/main/webapp/signup.js
Comment (wjiamin) [inline where-to-travel/src/main/webapp/signup.js]: Fix indentation here?

File: where-to-travel/src/main/webapp/signup.js
Comment (wjiamin) [inline where-to-travel/src/main/webapp/signup.js]: let ref = firebase.database().ref('users');

File: where-to-travel/src/main/webapp/signup.js
Comment (wjiamin) [inline where-to-travel/src/main/webapp/signup.js]: Inline this? :   } else if

File: where-to-travel/src/main/webapp/signup.js
Comment (wjiamin) [inline where-to-travel/src/main/webapp/signup.js:3]: Since most of these global constants are only used in 1 function, could we just define them in that function scope? Usually, avoid globals as much as possible.

**Review by wjiamin — COMMENTED** [review]
LGTM if all comments are resolved.

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js:2]: Since you could log in with a different email I don't think that this should be constant case. Jasmine sent me this link and I found it very helpful for understanding constant case. https://www.freecodecamp.org/news/when-to-capitalize-your-javascript-constants-4fabc0a4a4c4/

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js:3]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js:11]: Notifying the user is done in the validate function so there is an alert for that case.

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js:38]: good idea. done

File: where-to-travel/src/main/webapp/signup.js
Comment (wjiamin) [reply where-to-travel/src/main/webapp/signup.js:11]: It might be worth adding a comment about this.

**Review by wjiamin — COMMENTED** [review]
LGTM, will wait for Sailesh to approve.

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js:11]: ok done

File: where-to-travel/src/main/webapp/signup.js
Comment (saileshsimhadri) [inline where-to-travel/src/main/webapp/signup.js]: space between before and after ==

**Review by saileshsimhadri — APPROVED** [review]

File: where-to-travel/src/main/webapp/signup.js
Comment (emmapaczkowski) [reply where-to-travel/src/main/webapp/signup.js]: done

**Pull request merged**
